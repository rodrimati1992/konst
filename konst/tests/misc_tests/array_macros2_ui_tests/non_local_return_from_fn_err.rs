#[allow(unreachable_code)]
const _: [usize; 1] = 'out: {
    konst::array::from_fn_!(|i| break 'out [i * 2])
};


fn main() {}