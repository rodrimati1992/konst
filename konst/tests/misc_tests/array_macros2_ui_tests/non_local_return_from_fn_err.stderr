error[E0493]: destructor of `ArrayBuilder<usize, 1>` cannot be evaluated at compile-time
 --> tests/misc_tests/array_macros2_ui_tests/non_local_return_from_fn_err.rs:3:5
  |
3 |     konst::array::from_fn_!(|i| break 'out [i * 2])
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |     |
  |     the destructor for this type cannot be evaluated in constants
  |     value is dropped here
  |
  = note: this error originates in the macro `$crate::__array_map2__with_parsed_closure` which comes from the expansion of the macro `konst::array::from_fn_` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `ArrayConsumer<(), 1>` cannot be evaluated at compile-time
 --> tests/misc_tests/array_macros2_ui_tests/non_local_return_from_fn_err.rs:3:5
  |
3 |     konst::array::from_fn_!(|i| break 'out [i * 2])
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |     |
  |     the destructor for this type cannot be evaluated in constants
  |     value is dropped here
  |
  = note: this error originates in the macro `$crate::__array_map2__with_parsed_closure` which comes from the expansion of the macro `konst::array::from_fn_` (in Nightly builds, run with -Z macro-backtrace for more info)
